File: src/main/java/graphics/controls/Incrementer.java
package graphics.controls;

import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.control.Button;
import javafx.scene.text.Text;

import java.net.URL;
import java.util.ResourceBundle;

/**
 * Created by Mitch on 5/31/2017.
 */
public class Incrementer implements Initializable {
    private final int DEFAULT_VALUE = 6;

    private int value;

    @FXML private Button    dec_button;
    @FXML private Text      num_label;
    @FXML private Button    inc_button;


    public int get_value() {
        return value;
    }

    public void set_value(int value){
        this.value = value;
        num_label.setText(value+"");
    }

    @FXML
    private void increment() {
        set_value(value + 1);
    }

    @FXML
    private void decrement() {
        set_value(value - 1);
    }

    @Override
    public void initialize(URL location, ResourceBundle resources) {
        set_value(DEFAULT_VALUE);
    }
}

File: src/main/scala/graphical/controls/Incrementer.scala
package main.scala.graphical.controls

import scalafx.beans.property.IntegerProperty
import scalafx.scene.control.Button
import scalafx.scene.layout.{BorderPane, HBox}
import scalafx.scene.text.Text

/**
  * Created by Mitch on 7/30/2017.
  */
class Incrementer(val min_value: Int, val max_value: Int, def_value: Int) extends BorderPane {
  assert(min_value <= def_value && def_value <= max_value)

  val num_label = new Text(def_value.toString) {id = "num_label"}
  val value = IntegerProperty(def_value)
  value onChange {(_, _, n) => num_label.text = n.toString}

  def decrement(): Unit = {
    val m = this.value.get
    if (m > min_value)
      value.setValue(m - 1)
  }

  def increment(): Unit = {
    val m = this.value.get
    if (m < max_value)
      value.setValue(m + 1)
  }

  def set_value(v: Int): Unit =
    if (min_value < v && v < max_value)
      value.set(v)

  maxWidth  = Double.NegativeInfinity
  maxHeight = Double.NegativeInfinity
  prefWidth = 160.0
  stylesheets = Seq("@../../css/increment.css")

  left = new Button("-") {
    id = "dec_button"
    onMouseClicked = _ => decrement()
  }
  center = num_label
  right = new Button("+") {
    id = "inc_button"
    onMouseClicked = _ => increment()
  }
  styleClass = Seq("incrementer")
}
